scalyr_agent/configuration.py

```diff
@@ -238,7 +245,8 @@ def parse(self):
 
             # Set defaults based on `max_send_rate_enforcement` value
             if (
-                not self.__config["disable_max_send_rate_enforcement_overrides"]
+                self.__log_warnings
```

Comment:
This path includes the setting of the default values which we probably don't want to turn off along with the warnings, line 271 for example: `self.__config["max_allowed_request_size"] = 5900000`. If I remember correctly `_warn_of_override_due_to_rate_enforcement` only conditionally logs a warning so putting this check in there might be a better place. Setting these values probably won't matter in the case of requesting a status but better that the option did what it said on the tin (in that vein maybe `_warn_of_override_due_to_rate_enforcement` could do with a rename too).

Reply:
Yeah, like like you said - it shouldn't really matter since those settings really only apply to main agent process and have no affect on the status command itself.

Having said that, I can change it and put that logic in the __warn functions.

Tests would also be nice, but as mentioned above, for that we would need to end to end tests which will take more work.

Reply:
b559eefab6d3f83f10264c0cff9e66dc9f63a85a